docs/content/v0-upgrade.md

```diff
@@ -0,0 +1,471 @@
+upgrade gradually.
+
+This documentation covers the different upgrade scenarios and the best course of
+action for each. The documentation makes use of the following concepts:
+
+- **Bundle Producer**: A system based on `opa build` that produces a bundle that
+  is loaded by consumers.
+- **Bundle Consumer**: An OPA instance that loads and evaluates policy
+  from a bundle in the system.
+- **Authoring**: The process of writing Rego policies before bundles are
+  produced and consumed. In managed systems, this might be a user's only contact
+  point with OPA.
+
+In some systems, where OPA is used without a bundle, there is no producer. This
+simplifies the upgrade process.
+
+Users are encouraged to upgrade to OPA v1.0 as soon as possible. Using the v0
+compatible functionality until updating Rego is preferred to delaying the
+upgrade. The first part of this guide refers to upgrading OPA instances used for
+producing and consuming bundles. This is the first step users should take unless
+their Rego is already v1.0 compatible. See [Upgrading Rego](#upgrading-rego)
+below for information on how to upgrade Rego policies to be v1.0 compatible.
+
+## General Upgrade Approach: Upgrade Producers, then Consumers
+
+Users will need to upgrade to OPA v1.0 in their own way, depending on their
+release and change management processes, use cases and risk tolerance. The
+general advice is to upgrade producers first, then consumers. This is because
+the updated producers would be able to set the Rego version on bundle manifests
+and as a result it wouldn't be necessary to run consumers with the `--v0-compatible` flag.
+Also since it's likely there are much more consumers than producers, upgrading producers
+first would lead to a smoother upgrade process.
+
+Some users may wish to migrate to OPA v1.0 all at once, with adequate testing and validation
+this is possible. Not all steps are necessary for all users so a hybrid approach
+is also an option depending on your context.
+
+The rest of this documentation is designed to meet users where they find
+themselves and direct them down the smoothest path to upgrade to OPA
+v1.0.
+
+## Detailed Producer & Consumer Version Scenarios
+
+Tabulated in this section are the different versions of OPA users might be
+working with in different parts of their systems. Select the scenario that best
+matches your setup to find the recommended upgrade path.
+
+If you are in doubt, [Scenario 1](#scenario-1) is the most common starting
+point and we recommended you start there.
+
+|                           | v0.x Consumer                | Mix Consumer              | v1.0 Consumer                        |
+|---------------------------|--------------------------------------|---------------------------|--------------------------------------|
+| v0.x Producer             | [Scenario 1](#scenario-1) (All v0.x) | [Scenario 4](#scenario-4) | [Scenario 7](#scenario-7)            |
```

Comment:
Could we make the first column with producers bold, to make it look like a header like the consumer row.

Or is there some way to make the first column also be an actual table header?

Reply:
Not sure. Maybe @charlieegan3 has an idea on how to do this.